### PR TITLE
Adding sitecertificate resource type to Microsoft.Web RP

### PR DESCRIPTION
**Pull request by Gargimitra12300**
# Choose a PR Template

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

**Comment by ms-bogdan-rotaru** [comment]
Please fill in the purpose of this PR by choosing the correct PR template, and also acknowledge the due diligence checklist.

File: specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+            "description": "Name of the site.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
```
Comment (ms-bogdan-rotaru) [inline specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json:42]: Have you considered using the common types?

File: specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+          "List Certificates by resource group": {
+            "$ref": "./examples/ListSiteCertificatesByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{siteName}/certificates/{name}": {
```
Comment (ms-bogdan-rotaru) [inline specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json]: >name [](http://example.com/codeflow?start=127&length=4)

Maybe "certificateName" instead of simply "name"?

File: specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+            "$ref": "./examples/DeleteSiteCertificate.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "SiteCertificates"
+        ],
+        "summary": "Create or update a certificate under a given site.",
+        "description": "Description for Create or update a certificate.",
```
Comment (ms-bogdan-rotaru) [inline specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json:246]: Is this summary/description accurate?

File: specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+            "$ref": "./examples/DeleteSiteCertificateSlot.json"
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "SiteCertificates"
+        ],
+        "summary": "Create or update a certificate for a site and deployment slot.",
+        "description": "Description for Create or update a certificate.",
```
Comment (ms-bogdan-rotaru) [inline specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/SiteCertificates.json:558]: Is this accurate?

File: specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/CommonDefinitions.json [hunk 5]
```diff
@@ -673,6 +673,308 @@
...
+        {
+          "$ref": "#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "CertificatePatchResource resource specific properties",
+          "type": "object",
+          "properties": {
+            "password": {
```
Comment (ms-bogdan-rotaru) [inline specification/web/resource-manager/Microsoft.Web/stable/2024-04-01/CommonDefinitions.json:852]: >password [](http://example.com/codeflow?start=13&length=8)

Should set "x-ms-secret" ?

File: package-lock.json [hunk 6]
```diff
@@ -2256,6 +2256,7 @@
       "resolved": "https://registry.npmjs.org/prettier/-/prettier-3.2.5.tgz",
       "integrity": "sha512-3/GWa9aOC0YeD7LUfvOG2NiDyhOWRvt1k+rcKhOuYnMY24iiCphgneUfJDyFXd6rZCAnuLBv6UeAULtrhT/F4A==",
       "dev": true,
+      "license": "MIT",
```
Comment (mikeharder) [inline package-lock.json:2259]: Revert the changes to this file.  Also, I recommend updating your release branch with the latest commits from `main` (it's currently 722 commits behind).
